src/main.rs

```diff
@@ -30,65 +42,85 @@ async fn main() -> Result<(), Error> {
+    for (i, url) in args.urls.iter().enumerate() {
+        log::info!(
+            "{} Cloning {url:?}...",
+            format!("[{}/{}]", i + 1, url_count + 1).bold().blue()
+        );
+
+        let steps = 3 + (if args.zipped { 2 } else { 0 });
+
+        log::info!(
+            "{} {} Validating url...",
+            format!("[1/{steps}]").bold().yellow(),
             output::LOOKING_GLASS
         );
 
         let path = match parser::parse_url(url) {
             Ok(path) => path,
             Err(err) => {
-                eprintln!("{}", err.to_string().red());
+                log::error!("{}", err.to_string().red());
                 process::exit(0);
             }
         };
 
         let data = match parser::parse_path(&path, args.path.clone()) {
             Ok(data) => data,
             Err(err) => {
-                eprintln!("{}", err.to_string().red());
+                log::error!("{}", err.to_string().red());
                 process::exit(0);
             }
         };
 
-        println!(
-            "{} {}Downloading...",
-            "[2/3]".bold().yellow(),
+        log::info!(
+            "{} {} Downloading...",
+            format!("[2/{steps}]").bold().yellow(),
             output::TRUCK
         );
 
         match requests::fetch_data(&data).await {
             Err(err) => {
-                eprintln!("{}", err.to_string().red());
+                log::error!("{}", err.to_string().red());
                 process::exit(0);
             }
-            Ok(_) => println!(
-                "{} {}Downloaded Successfully.",
-                "[3/3]".bold().yellow(),
+            Ok(_) => log::info!(
+                "{} {} Downloaded successfully.",
+                format!("[3/{steps}]").bold().yellow(),
                 output::SPARKLES
             ),
         };
 
         if args.zipped {
+            log::info!(
+                "{} {} Zipping...",
+                format!("[4/{steps}]").bold().yellow(),
+                output::PACKAGE
+            );
+
             let dst_zip = format!("{}.zip", &data.root);
             let zipper = ZipArchiver::new(&data.root, &dst_zip);
             match zipper.run() {
-                Ok(_) => (),
+                Ok(_) => log::info!(
+                    "{} {} Zipped successfully.",
+                    format!("[5/{steps}]").bold().yellow(),
+                    output::SPARKLES
+                ),
                 Err(ZipError::FileNotFound) => {
-                    eprintln!("{}", "\ncould not zip the downloaded file".bold().red())
+                    log::error!("{}", "Failed to zip files".bold().red())
                 }
-                Err(e) => eprintln!("{}", e.to_string().bold().red()),
+                Err(e) => log::error!("{}", e.to_string().bold().red()),
             }
         }
     }
 
-    println!(
-        "
-        [+] Downloaded {:?} dir(s).",
-        &args.urls.len()
+    log::info!(
+        "{} Downloaded {:?} director{}.",
```

Comment:
I should note that this is technically inaccurate some of the time since individual files (not just directories) can be downloaded.